src/main/kotlin/org/opensearch/replication/action/replicationstatedetails/TransportUpdateReplicationStateDetails.kt

```diff
@@ -43,19 +44,24 @@ class TransportUpdateReplicationStateDetails @Inject constructor(transportServic
                 transportService, clusterService, threadPool, actionFilters, ::UpdateReplicationStateDetailsRequest, indexNameExpressionResolver),
         CoroutineScope by GlobalScope {
 
+    companion object {
+        const val DEFAULT_TIMEOUT = 60000L // Twice the default cluster publish timeout
+        const val DEFAULT_TIMEOUT_FACTOR = 3L // Thrice the current updated settings
+    }
     override fun checkBlock(request: UpdateReplicationStateDetailsRequest, state: ClusterState): ClusterBlockException? {
         return state.blocks.globalBlockedException(ClusterBlockLevel.METADATA_WRITE)
     }
 
     override fun masterOperation(request: UpdateReplicationStateDetailsRequest, state: ClusterState,
                                  listener: ActionListener<AcknowledgedResponse>) {
+        val clusterPublishTimeout = state.metadata.settings().getAsLong(Coordinator.PUBLISH_TIMEOUT_SETTING.key, DEFAULT_TIMEOUT)
```

Comment:
Should we leverage listener to capture settings in local variable? (Or does this not have any perf impact?)